src/number.cr

```diff
@@ -71,6 +71,13 @@ struct Number
   # ints = Int64[1, 2, 3]
   # ints.class # => Array(Int64)
   # ```
+  #
+  # This is equivalent to an array literal of the same item type:
```

Comment:
The macro uses `new`, so the two are not equivalent: https://github.com/crystal-lang/crystal/issues/13021#issuecomment-1409359263

Reply:
Let's call it similar then?
```suggestion
  # This is similar to an array literal of the same item type:
```